crates/types/src/config/common.rs

```diff
@@ -233,6 +233,11 @@ pub struct CommonOptions {
     #[serde(with = "serde_with::As::<serde_with::DisplayFromStr>")]
     #[cfg_attr(feature = "schemars", schemars(with = "String"))]
     pub metadata_update_interval: humantime::Duration,
+
+    /// # Network error retry policy
+    ///
+    /// The retry policy for node network error
+    network_error_retry_policy: Option<RetryPolicy>,
```

Comment:
Wondering whether we can define the type as `RetryPolicy` because `RetryPolicy` has a `None` variant as well.

Reply:
@tillrohrmann 
True, there is a None variant on the RetryPolicy. But i didn't want break compatibility with current configurations files which has no network_error_retry_policy defined. If we just use RetryPolicy it will default to None (hence no retry), which breaks the default behaviour which has exponential retry.

But with Option, if network_error_retry_policy is completely missing from the config the function `network_error_retry_policy` can still return a default policy (the one that was hard coded in code), unless you configure retry policy explicitly to None in config (like `network_error_retry_policy: none` ) to explicitly disable retries

Reply:
Can we configure serde using the same default value as the one we define in the `Default` implementation? Then if nothing is specified we would default to 
```RetryPolicy::exponential(
      Duration::from_millis(10),
      2.0,
      Some(15),
      Some(Duration::from_secs(5))
```

Reply:
I think that the current `Default` implementation should already be respected. No need to instruct `serde` separately.

Reply:
Ah that's good to know. Good one :)